### PR TITLE
Move {RAsm,RAnal}::addrbytes to RIO::addrbytes, delete asm.addrbytes and make cLEMENCy work again

### PR DESCRIPTION
**Pull request by MaskRay**
`RAsm::user` in rasm2.c does not point to `RCore`, thus I cannot access `core->io->addrbytes` in rasm2.c 
radare2.c should be good.

### CODE REVIEW

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -353,7 +353,7 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 		0
 	};
 	ut64 ret = UT64_MAX;
-	for (offs = 0; offs < len; offs += anal->addrbytes * oplen) {
+	for (offs = 0; offs < len; offs += anal->iob.io->addrbytes * oplen) {
```
Comment (radare) [inline libr/anal/fcn.c]: asuming io is not null and accessing all those pointers of pointers on each iteration is unnecessary hold that value before entering the loop in a const int which defaults to 1 if io is null

File: libr/asm/asm.c [hunk 2]
```diff
@@ -438,7 +439,8 @@ R_API int r_asm_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	}
 	//XXX check against R_ASM_BUFSIZE other oob write
 	memcpy (op->buf, buf, R_MIN (R_ASM_BUFSIZE - 1, oplen));
-	r_hex_bin2str (buf, R_MIN (a->addrbytes * oplen,
+	int addrbytes = a->user ? ((RCore *)a->user)->io->addrbytes : 1;
```
Comment (radare) [inline libr/asm/asm.c]: const

Comment (MaskRay) [reply libr/asm/asm.c]: Done

File: libr/anal/fcn.c (hunk 1 above)
Comment (MaskRay) [reply libr/anal/fcn.c]: Can you elaborate?

File: libr/anal/fcn.c [hunk 3]
```diff
@@ -353,7 +353,8 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 		0
 	};
 	ut64 ret = UT64_MAX;
-	for (offs = 0; offs < len; offs += anal->addrbytes * oplen) {
+	const int addrbytes = anal->iob.io->addrbytes;
```
Comment (radare) [inline libr/anal/fcn.c]: And the null chk for io? Those apis can be used standalone we thout rcore and rio

File: libr/anal/fcn.c [hunk 4]
```diff
@@ -548,9 +549,9 @@ static int walk_switch(RAnal *anal, RAnalFunction *fcn, ut64 from, ut64 at) {
 }
 
 static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut64 len, int depth) {
-	int continue_after_jump = anal->opt.afterjmp;
-	int noncode = anal->opt.noncode;
-	int addrbytes = anal->addrbytes;
+	const int continue_after_jump = anal->opt.afterjmp;
+	const int noncode = anal->opt.noncode;
+	const int addrbytes = anal->iob.io->addrbytes;
```
Comment (radare) [inline libr/anal/fcn.c]: Null deref

File: libr/anal/fcn.c (hunk 3 above)
Comment (MaskRay) [reply libr/anal/fcn.c]: Added

File: libr/anal/fcn.c (hunk 4 above)
Comment (MaskRay) [reply libr/anal/fcn.c]: Added

**Pull request merged**